test/test_hydrostatic_free_surface_immersed_boundaries_congrad_solve.jl

```diff
@@ -0,0 +1,99 @@
+using Oceananigans.ImmersedBoundaries: ImmersedBoundaryGrid, GridFittedBottom
+using Oceananigans.Architectures: arch_array
+using Oceananigans.TurbulenceClosures: VerticallyImplicitTimeDiscretization
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: compute_vertically_integrated_volume_flux!
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: compute_implicit_free_surface_right_hand_side!
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: pressure_correct_velocities!
+
+@testset "Immersed boundaries test divergent flow solve with hydrostatic free surface models" begin
+    @info "Testing immersed boundaries divergent flow solve"
+
+    for arch in archs
+        Nx = 11 
+        Ny = 11 
+        Nz = 1
+
+        underlying_grid = RectilinearGrid(arch,
+                                          size = (Nx, Ny, Nz),
+                                          extent = (Nx, Ny, 1),
+                                          topology = (Periodic, Periodic, Bounded))
```

Comment:
This test is failing on `main` because we are missing `halo = (3, 3, 3)`. @simone-silvestri @christophernhill maybe we need to update how the CI works so this doesn't happen?